src/Event/EventBundle/Controller/Backend/MediaController.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+namespace Event\EventBundle\Controller\Backend;
+
+use Symfony\Component\Config\Definition\Exception\Exception;
+use Symfony\Component\HttpFoundation\Request;
+use Event\EventBundle\Controller\Controller;
+use Event\EventBundle\Entity\Media;
+use Event\EventBundle\Form\Type\MediaType;
+
+class MediaController extends Controller
+{
+
+    public function getUploadPath()
+    {
+        return $this->get('kernel')->getRootDir() . DIRECTORY_SEPARATOR . '..' . DIRECTORY_SEPARATOR .
+            'web' . DIRECTORY_SEPARATOR . 'uploads' . DIRECTORY_SEPARATOR . 'media' . DIRECTORY_SEPARATOR;
+    }
+    
+    public function getUploadUrl(){
+        return '/uploads/media/';
+    }
+
+    public function indexAction()
+    {
+        return $this->render('EventEventBundle:Backend/Media:index.html.twig', array(
+            'media' => $this->getRepository('EventEventBundle:Media')->findAll(),
+            'uploadPath' => $this->getUploadPath(),
+            'uploadUrl' => $this->getUploadUrl()
+        ));
+    }
+
+    public function manageAction(Request $request, $id = null)
+    {
+        $oldFileName = '';
+        if ($id === null) {
+            $entity = new Media();
+        } else {
+            $entity = $this->findOr404('EventEventBundle:Media', $id);
+            $oldFileName = $entity->getFilename();
+        }
+
+        $form = $this->createForm(MediaType::class, $entity);
+
+        if ($request->getMethod() === 'POST') {
+            $form->handleRequest($request);
+
+            if ($form->isValid()) {
+                if(is_null($entity->getId())){
+                    $entity->setCreatedDate(new \DateTime());
+                }
+
+                if($oldFileName !== $entity->getFilename()) {
+                    $url = $request->request->get('file_url');
+                    $path = $this->getUploadPath();
+                    $entity->setFilename( time() . '_' . $entity->getFilename());
+                    $file = $this->getFile($url);
+                    file_put_contents($path . $entity->getFilename(), $file);
+                    chmod($path . $entity->getFilename(), 0777);
+                    if(is_file($path . $oldFileName)) {
+                        unlink($path . $oldFileName);
+                    }
+                }
+
+                $entity->setUpdatedDate(new \DateTime());
+                $this->getManager()->persist($entity);
```

Comment:
use proxy methods from https://github.com/pilot/eventator/blob/master/src/Event/EventBundle/Controller/Controller.php

please take a look at the https://github.com/pilot/eventator/blob/master/src/Event/EventBundle/Controller/EventController.php
to understand the code structure